### PR TITLE
adding DRAGEN compatibility for fastq processing

### PR DESCRIPTION
**Pull request by uwidocki**
please review changes made for DRAGEN compatibility for processing fastq files

### CODE REVIEW

File: R/write_df_from_fastq.R [hunk 1]
```diff
@@ -113,3 +113,99 @@ write_df_from_fastq <- function(
...
+      
+      forward_reads_cl_barcode = as.character(forward_reads_cl_barcode)
+      
+      # get the index 1 and index 2 barcodes from the fastq file
+      print("Get index 1 and index 2 barcodes from fastq file")
+      
+      # do the below string splitting but for each read in the fastq file
+      temp <- as.character(forward_reads_chunk@id)
+      
+      fastq_header_split <- function(str){
```
Comment (AnupJonchhe) [inline R/write_df_from_fastq.R:164]: No need for the Sample Sheet then? Is this assuming that the file name contains the Index1 and Index2 then

File: scripts/fastq2readcount.R [hunk 2]
```diff
@@ -80,15 +88,27 @@ if (is.null(args$write_interval)){
...
-                                  index_1_file = index_1_files,
-                                  index_2_file = index_2_files,
-                                  write_interval = write_interval, 
-                                  PLATE_BC_LENGTH = read_lengths[1],
-                                  WELL_BC_LENGTH = read_lengths[2],
-                                  CL_BC_LENGTH = read_lengths[3],
-                                  save_loc = args$out
-                        )
+
+# here,
```
Comment (AnupJonchhe) [inline scripts/fastq2readcount.R:92]: Good approach for backwards compatibility!

File: R/write_df_from_fastq.R (hunk 1 above)
Comment (AnupJonchhe) [reply R/write_df_from_fastq.R:164]: If so, might just want to add that as a comment, otherwise this looks good to me

**Review by maha-naim — COMMENTED** [review]
Looks good!

**Pull request merged**